### PR TITLE
Google analytics

### PR DESCRIPTION
**Pull request by chu4**
Added global analytics on production env

**CHECK LIST**
- [ ] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [ ] screenshot of rails_best_practices and other tools added
- [ ] screenshot of feature is added if task has UI

### CODE REVIEW

File: app/views/layouts/shared/_google_analytics.html.slim [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- if Rails.env == "production"
+  /! Global site tag (gtag.js) - Google Analytics
+  script async="" src="https://www.googletagmanager.com/gtag/js?id=UA-112609815-1"
```
Comment (romaslmd) [inline app/views/layouts/shared/_google_analytics.html.slim]: move ID to secrets

File: app/views/layouts/landing.html.slim [hunk 2]
```diff
@@ -1,6 +1,8 @@
 doctype html
 html
   head
+    = render 'layouts/shared/google_analytics'
```
Comment (romaslmd) [inline app/views/layouts/landing.html.slim]: could ya create a helper?

```
def render_google_analytics
  return unless Rails.env.production?
  render 'layouts/shared/google_analytics', id: Rails.secrets.ga
end
```

File: app/views/layouts/application.html.slim [hunk 3]
```diff
@@ -1,6 +1,8 @@
 doctype html
 html
   head
+    = render 'layouts/shared/google_analytics'
```
Comment (romaslmd) [inline app/views/layouts/application.html.slim]: keep JS scripts with JS scripts within head

Comment (igorkasyanchuk) [reply app/views/layouts/application.html.slim]: it's ok here, but just change to `render_google_analytics` how Roman suggested

File: app/views/layouts/shared/_google_analytics.html.slim (hunk 1 above)
Comment (igorkasyanchuk) [reply app/views/layouts/shared/_google_analytics.html.slim]: actually don't move to secrets.
this is not a private info.

BUT
do not convert it to slim

use ERB with same code as you got from google

**Comment by chu4** [comment]
Added Google Analytics Helper
Changed partial from slim to erb
structured javascripts together in layouts

**Pull request merged**